pypesto/objective/amici/amici.py

```diff
@@ -599,8 +603,13 @@ def set_custom_timepoints(
                 for _ in range(len(amici_objective.edatas))
             ]
 
-        amici_objective.custom_timepoints = custom_timepoints
-        amici_objective.apply_custom_timepoints()
+        if type(amici_objective.calculator) is HierarchicalAmiciCalculator:
```

Comment:
This will fail on instances of derived classes of `HierarchicalAmiciCalculator` - use `isinstance` instead.